src/components/icon/icon-close.component.js

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Events from '../../utils/helpers/events';
+import Icon from './icon.component';
+
+const IconClose = React.forwardRef(({
+  onClick: onClickProp,
+  onKeyDown: onKeyDownProp,
+  className,
+  children,
+  ...rest
+}, ref) => {
+  const onKeyDown = useCallback((e) => {
+    if (Events.isEnterKey(e) || Events.isSpaceKey(e)) {
+      e.preventDefault();
+      onKeyDownProp(e);
+    } else {
+      e.stopPropagation();
+    }
+  }, [onKeyDownProp]);
+
+  const onClick = useCallback((e) => {
+    onClickProp(e);
+  });
+
+  const extraProps = {
+    ...rest,
+    ref,
+    className: (className && classNames(className, 'close-icon')) || 'close-icon',
```

Comment:
do we still need this `close-icon` class? The component should be styled with `styled-components`.